scripts/add_electricity.py

```diff
@@ -622,6 +623,32 @@ def attach_extendable_generators(n, costs, ppl):
             )
 
 
+def attach_custom_renewables(n, tech_map, ppl):
```

Comment:
if the routine should work as for the conventionals, then the custom ppls would be added to the powerplants.csv aleady in build_powerplants.py and added to the network in the attach_wind_and_solar() function here. This would be more consistent (in terms of treatment of RE and conventionals), but would require additional changes.

Reply:
Yes, I understand. Nevertheless, if done cleanly, we should add it as you suggested in the function `attach_wind_and_solar(ppls)`. The function should be adapted such that it can also add `p_nom`, not only `p_nom_max`.  The custom renewable ppls should be added to `powerplants.csv` using the `add_custom_powerplants` function in build_powerplants. `estimate_renewbale_capacities_from_irena` should then only estimate renewables for countries where there are no renewables.